framework/PostProcessors/EconomicRatio.py

```diff
@@ -0,0 +1,830 @@
+    # median needs                 |                        | sortinoRatio, gainLossRatio
+    # sigma needs                  | variance               | sharpeRatio
+    # variance                     | expectedValue          | sigma
+    # expectedValue                |                        | sharpeRatio, sortinoRatio, gainLossRatio
+    # sharpeRatio needs            | expectedValue,sigma    |
+    # sortinoRatio needs           | expectedValue,median   |
+    # gainLossRatio needs          | expectedValue,median   |
+
+
+
+
+    # update needed dictionary when standard errors are requested
+    needed['expectedValue']['targets'].update(needed['sigma']['targets'])
+    needed['expectedValue']['targets'].update(needed['variance']['targets'])
+    needed['expectedValue']['targets'].update(needed['median']['targets'])
+    needed['expectedValue']['targets'].update(needed['sharpeRatio']['targets'])
+    needed['expectedValue']['targets'].update(needed['sortinoRatio']['targets'])
+    needed['expectedValue']['targets'].update(needed['gainLossRatio']['targets'])
+
+    needed['sigma']['targets'].update(needed['sharpeRatio']['targets'])
+
+    needed['variance']['targets'].update(needed['sigma']['targets'])
+
+
+
+    needed['median']['targets'].update(needed['sortinoRatio']['targets'])
+    needed['median']['targets'].update(needed['gainLossRatio']['targets'])
+
+
+    for metric, params in needed.items():
+      needed[metric]['targets'] = list(params['targets'])
+      try:
+        needed[metric]['features'] = list(params['features'])
+      except KeyError:
+        pass
+
+    #
+    # BEGIN actual calculations
+    #
+
+    calculations = {}
+
+    #################
+    # SCALAR VALUES #
+    #################
+    #
+    # expected value
+    #
+    metric = 'expectedValue'
+    if len(needed[metric]['targets']) > 0:
+      self.raiseADebug('Starting "'+metric+'"...')
+      dataSet = inputDataset[list(needed[metric]['targets'])]
+      if self.pbPresent:
+        relWeight = pbWeights[list(needed[metric]['targets'])]
+        equivalentSize = self.__computeEquivalentSampleSize(relWeight)
+        dataSet = dataSet * relWeight
+        expectedValueDS = dataSet.sum(dim = self.sampleTag)
+        calculations['equivalentSamples'] = equivalentSize
+      else:
+        expectedValueDS = dataSet.mean(dim = self.sampleTag)
+      calculations[metric] = expectedValueDS
+    #
+    # variance
+    #
+    metric = 'variance'
+    if len(needed[metric]['targets'])>0:
+      self.raiseADebug('Starting "'+metric+'"...')
+      dataSet = inputDataset[list(needed[metric]['targets'])]
+      meanSet = calculations['expectedValue'][list(needed[metric]['targets'])]
+      relWeight = pbWeights[list(needed[metric]['targets'])] if self.pbPresent else None
+      varianceDS = self._computeVariance(dataSet,meanSet,pbWeight=relWeight,dim=self.sampleTag)
+      calculations[metric] = varianceDS
+    #
+    # sigma
+    #
+    metric = 'sigma'
```

Comment:
you need to use the Basic Statistic PP...not replicate the calculation of these moments...extremely redundant